.github/workflows/test.yml

```diff
@@ -97,7 +100,26 @@ jobs:
       - name: cargo test
         uses: actions-rs/cargo@v1
         if: ${{ !matrix.cross }}
+        env:
+          NEXTEST_PROFILE: ci   # defined in .config/nextest.toml
         with:
-          command: test
-          args:  --workspace --target=${{ matrix.target }}
+          command: nextest
+          args:  run --workspace --target=${{ matrix.target }}
           use-cross: ${{ matrix.cross }}
+
+      - name: upload test results
+        uses: actions/upload-artifact@v3
+        if: ${{ !matrix.cross && ( success() || failure() ) }}
+        with:
+          name: Test results (${{ matrix.target }})
+          path: target/nextest/ci/junit.xml
+
+  event_file:
+    name: "Event File"
+    runs-on: ubuntu-latest
+    steps:
+    - name: Upload
+      uses: actions/upload-artifact@v2
+      with:
+        name: Event File
+        path: ${{ github.event_path }}
```

Comment:
If i understand this correctly, this uploads the full webhook event, that would be send if there were any webhooks specified?
This is then unpacked in the test-report action and used in the publish-unit-test-result-action call.

It would be great to have a few lines of docs on why this is desirable.

Reply:
Again, it's a requirement set by the publishing action, as it uses the event file as a basis. The separate workflow handling the reports means it would otherwise have the wrong context.